### PR TITLE
Update REST API reference with Batch API details

### PR DESCRIPTION
**Pull request by marklewin**
> _NOTE TO REVIEWERS_ - please **do not review PRs in the `DRAFT` state**, as the PR may change substantially before it is ready to review. Thanks.

## Description

* Epic: [DOC-4](https://ably.atlassian.net/browse/DOC-4)
* Task: [DOC-493](https://ably.atlassian.net/browse/DOC-493)

## Review

Please review the Batch section of the REST API reference:

* [Batch API ref](https://ably-docs-pr-1231.herokuapp.com/rest-api/#batch)

### CODE REVIEW

File: content/rest-api/index.textile [hunk 1]
```diff
@@ -1216,6 +1219,147 @@ Stats records contain a hierarchy of elements relating to messages, connections
...
+
+bc[json]. {
+  channels: <channel names>,
+  messages: <messages>
+}
+
+* @<channel names>@ is a single channel name, or an array of up to 100 channel names, expressed as strings.
+* @<messages>@ is a single message, or an array of messages.
+
+The maximum size of each request body must be less than 2MiB and the total size of all messages in a @messages@ array must be less than the "message size limit":https://faqs.ably.com/what-is-the-maximum-message-size (64kiB for paid accounts, 16kiB for free accounts).
```
Comment (paddybyers) [inline content/rest-api/index.textile:1242]: 64kiB by default for paid accounts

File: content/rest-api/index.textile [hunk 2]
```diff
@@ -1216,6 +1219,147 @@ Stats records contain a hierarchy of elements relating to messages, connections
...
+        "messageId": "3PodoYLYlR:0"
+    },
+    {
+        "channel": "channel3",
+        "messageId": "TMefwp6-lP:0"
+    }
+]
+
+When unsuccessful or partially unsuccessful, the response includes an "ErrorInfo":/rest/types#error-info object.
+
```
Comment (paddybyers) [inline content/rest-api/index.textile:1290]: extra whitespace

File: content/rest-api/index.textile [hunk 3]
```diff
@@ -1216,6 +1219,147 @@ Stats records contain a hierarchy of elements relating to messages, connections
...
+            "action":1
+         }
+      ]
+   }
+]
+
+Each object in the @presence@ array for each channel describes the presence state of one of that channel's members:
+
+bc[json]. [{
+  id: <a unique member identifer generated by Ably>,
```
Comment (paddybyers) [inline content/rest-api/index.textile:1353]: identifier

File: content/rest-api/index.textile (hunk 2 above)
Comment (marklewin) [reply content/rest-api/index.textile:1290]: Fixed in [a916c43](https://github.com/ably/docs/pull/1231/commits/a916c432f94f68787328fdf3da4a114c0e6de456).

File: content/rest-api/index.textile (hunk 1 above)
Comment (marklewin) [reply content/rest-api/index.textile]: Fixed in [a916c43](https://github.com/ably/docs/pull/1231/commits/a916c432f94f68787328fdf3da4a114c0e6de456).

**Review by paddybyers — APPROVED** [review]
LGTM, thanks

File: content/rest-api/index.textile [hunk 4]
```diff
@@ -1216,6 +1219,146 @@ Stats records contain a hierarchy of elements relating to messages, connections
...
+
+bc[text]. {
+  channels: <channel names>,
+  messages: <messages>
+}
+
+* @<channel names>@ is a single channel name, or an array of up to 100 channel names, expressed as strings.
+* @<messages>@ is a single message, or an array of messages.
+
+The maximum size of each request body must be less than 2MiB and the total size of all messages in a @messages@ array must be less than the "message size limit":https://faqs.ably.com/what-is-the-maximum-message-size (64kiB by default for paid accounts, 16kiB for free accounts).
```
Comment (m-hulbert) [inline content/rest-api/index.textile:1242]: We could update the message size limit link to: `/general/limits#message-size` and avoid linking out of the docs.

File: content/rest-api/index.textile [hunk 5]
```diff
@@ -1216,6 +1219,146 @@ Stats records contain a hierarchy of elements relating to messages, connections
...
+        "channel": "channel2",
+        "messageId": "3PodoYLYlR:0"
+    },
+    {
+        "channel": "channel3",
+        "messageId": "TMefwp6-lP:0"
+    }
+]
+
+When unsuccessful or partially unsuccessful, the response includes an "ErrorInfo":/rest/types#error-info object.
```
Comment (m-hulbert) [inline content/rest-api/index.textile:1289]: I think we should align on terminology here, as the batch topic refers to success, failure and partial success. What do you think?

File: content/rest-api/index.textile [hunk 6]
```diff
@@ -1216,6 +1219,146 @@ Stats records contain a hierarchy of elements relating to messages, connections
...
+
+h3(#batch-presence). Batch presence
+
```
[the remark this answers is not in the document]
Comment (marklewin) [reply content/rest-api/index.textile:1294]: Good spot! Fixed in [9c10c4e](https://github.com/ably/docs/pull/1231/commits/9c10c4e9f414030eb2a6423a3f99ad95f5b557ac).

File: content/rest-api/index.textile (hunk 4 above)
Comment (marklewin) [reply content/rest-api/index.textile]: Good idea. Addressed in [9c10c4e](https://github.com/ably/docs/pull/1231/commits/9c10c4e9f414030eb2a6423a3f99ad95f5b557ac).

File: content/rest-api/index.textile (hunk 5 above)
Comment (marklewin) [reply content/rest-api/index.textile]: I agree. Please see proposed change in [9c10c4e](https://github.com/ably/docs/pull/1231/commits/9c10c4e9f414030eb2a6423a3f99ad95f5b557ac).

**Review by m-hulbert — APPROVED** [review]
LGTM!

**Pull request merged**